javascript/src/main/components/BasicCourseSearch/CourseSearchCourseStartEndQtr.js

```diff
@@ -0,0 +1,180 @@
+import React, { useState } from "react";
+import { Form, Button } from "react-bootstrap";
+
+const CourseSearchCourseStartEndQtr = ({ setCourseJSON, fetchJSON }) => {
+
+    const [startQuarter, setStartQuarter] = useState("20211");
+    const [endQuarter, setEndQuarter] = useState("20211");
+    const [subjectArea, setSubjectArea] = useState("CMPSC   ");
+    const [courseNumber, setCourseNumber] = useState("");
+    const [courseSuf, setCourseSuf] = useState("");
+
+    const handleSubmit = (event) => {
+        //console.log(event);
+        event.preventDefault();
+        fetchJSON(event, { startQuarter, endQuarter, subjectArea, courseNumber, courseSuf}).then((courseJSON) => {
+            setCourseJSON(courseJSON);
+        });
+    };
+
+    // FALL = 4, F
+    // WINTER = 1, W
+    // SPRING = 2, S
+    // SUMMER = 3, M
+    const handleStartQuarterOnChange = (event) => {
+        setStartQuarter(event.target.value);
+    };
+
+    const handleEndQuarterOnChange = (event) => {
+        setEndQuarter(event.target.value);
+    };
+
+    const handleSubjectAreaOnChange = (event) => {
+        setSubjectArea(event.target.value);
+    };
+
+    const handleCourseNumberOnChange = (event) => {
+        setCourseNumber(event.target.value);
+    }
+
+    const handleCourseSufOnChange = (event) => {
+        setCourseSuf(event.target.value);
+    }
+    // Note: Not all possible courses were able to be added in the subject area list as many of them were dead links that provided no information
+    return (
+        <Form onSubmit={handleSubmit}>
+            <Form.Group controlId="CourseNameSearch.StartQuarter">
+                <Form.Label>Start Quarter</Form.Label>
+                <Form.Control as="select" onChange={handleStartQuarterOnChange} value={startQuarter}  >
+                    <option value="20211">W21</option>
+
+                    <option value="20204">F20</option>
+                    <option value="20201">W20</option>
+                    <option value="20202">S20</option>
+                    <option value="20203">M20</option>
+
+                </Form.Control>
+            </Form.Group>
+            <Form.Group controlId="CourseNameSearch.EndQuarter">
+                <Form.Label>End Quarter</Form.Label>
+                <Form.Control as="select" onChange={handleEndQuarterOnChange} value={endQuarter}  >
+                    <option value="20211">W21</option>
+
```

Comment:
any reason for the empty lines on lines 50, 55, and 62? if not we should probably remove these.